docs/comlink/reference/binary_data.mdx

```diff
@@ -0,0 +1,201 @@
+
+### Send file in body
+```
+map SendFile {
+  http POST "/resource" {
+    request "application/octet-stream" {
+      body = input.file
+    }
+
+    response 200 "text/plain" {
+      map result body
+    }
+  }
+}
+```
+
+### Send multiple files as FormData field alongside other fields
+```
+map SendFile {
+  http POST "/resource" {
+    request "multipart/form-data" {
+      body {
+        field = input.field
+        file1 = input.file1
+        file2 = input.file2
+      }
+    }
+
+    response 200 "text/plain" {
+      map result body
+    }
+  }
+}
+```
+
+### Upload file by chunks
+```
+profile = "example@1.0"
+provider = "localhost"
+
+map ChunkedUpload {
+  uploadResults = call foreach(chunk of input.file.chunkBy(10)) Upload(chunk = chunk)
+
+  return map result uploadResults
+}
+
+operation Upload {
+  http POST "/upload/{args.uploadId}" {
+    request "application/octet-stream" {
+      body = args.chunk
+    }
+
+    response 200 "text/plain" {
+      return true
+    }
+
+    response 500 "text/plain" {
+      return false
+    }
+  }
+}
+```
+
+### Read first bytes of file and then send it in body
+```
+map SendFile {
+  firstTen = input.file.peek(10)
+
+  contentType = firstTen.toString('ascii').toLowerCase().includes('pdf') ? 'application/pdf' : 'application/octet-stream'
+
+  http POST "/resource" {
+    request {
+      headers {
+        'content-type' = contentType
+      }
+      
+      body = input.file
+
+    response 200 "text/plain" {
+      map result body
+    }
+  }
+}
+```
+
+### Update BinaryData from Comlink Map
+```
+map Example {
+  tmp = (input.file.name = input.filename)
+  tmp = (input.file.mimetype = input.mimetype)
+  
+  http POST "/" {
+    request "multipart/form-data" {
+      body = {
+        file: input.file
+      }
+    }
+  
+    response 200 {
+      map result body
+    }
+  }
+}
+```
+
+## OneSDK use example
+
+```typescript
+import { SuperfaceClient, BinaryData } from '@superfaceai/one-sdk';
+
+const client = new SuperfaceClient();
+const profile = await client.getProfile('example');
+
+// for files
+const result = await profile.getUseCase('Example').perform({
+  binary: BinaryData.fromPath(pathToFile, { mimetype: 'text/plain', filename: 'example.txt' });
+
+// or for Readable streams
+const stream = getReadableStream();
+const result = await profile.getUseCase('Example').perform({
+  binary: BinaryData.fromStream(stream);
+});
+```
+
+## Limitations
+
+### 1. Asynchronous operations on BinaryData
+
+Current limitation for use in Comlink Map is that `peek`, `read` and `getAllData` methods are asynchronous. So, they can't be directly chained with function on [Buffer](https://nodejs.org/dist/latest-v18.x/docs/api/buffer.html#class-buffer). Most likely `toString()`.
+
+To overcome this limitation, It must be first [set into variable](https://spec.superface.dev/latest/map-spec.html#sec-Set-Variables), so the interpreter resolves the Promise:
```

Comment:
```suggestion
To overcome this limitation, `BinaryData` must first be [assigned to a variable](https://spec.superface.dev/latest/map-spec.html#sec-Set-Variables), so that the interpreter resolves the Promise:
```